odxtools/statechart.py

```diff
@@ -0,0 +1,93 @@
+# SPDX-License-Identifier: MIT
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+from .odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkId
+from .state import State
+from .statetransition import StateTransition
+from .utils import create_description_from_et
+
+
+@dataclass
+class StateChart:
+    """
+    Corresponds to STATE-CHART.
+    """
+
+    odx_id: OdxLinkId
+    short_name: str
+    long_name: Optional[str]
+    description: Optional[str]
+    semantic: str
+    state_transitions: List[StateTransition]
+    start_state_snref: str
+    states: List[State]
+
+    @property
+    def start_state(self) -> State:
+        return self._start_state
+
+    @staticmethod
+    def from_et(et_element, doc_frags: List[OdxDocFragment]) -> "StateChart":
+        short_name = et_element.findtext("SHORT-NAME")
+        odx_id = OdxLinkId.from_et(et_element, doc_frags)
+        assert odx_id is not None
+
+        long_name = et_element.findtext("LONG-NAME")
+        description = create_description_from_et(et_element.find("DESC"))
+
+        semantic = et_element.findtext("SEMANTIC")
+        assert semantic is not None
```

Comment:
the code does not depend on the value of this field, maybe a warning is enough?

Reply:
the XSD says that the `SEMANTIC` tag is mandatory. you're right that this should not be an plain `assert`, tough...